Services/src/main/resources/webutils/templates/webutils-templates.html

```diff
@@ -1358,7 +1362,104 @@ <h4 class="modal-title">${attr('title')}</h4>
 			}, {"element": element, "$scope" : context.$scope}));
 		</post-script>
 	</directive>
+	
+	<directive name="ykColorField" required-attr="name, ng-model">
+		<content>
+			<set-var name="_readOnly" 
+				expr="context.attr('yk-read-only')? context.attr('yk-read-only') : 'false'"/>
+			<set-var name="_onChange" 
+				expr="context.attr('ng-change')? context.attr('ng-change') : ''"/>
+			<set-var name="_placeholder" 
+				expr="context.attr('placeholder')? context.attr('placeholder') : ''"/>
 
+			<div class="form-group" style="margin-bottom: 0px;">
+				<div class="input-group colorpicker-component" style="width: 10em;">
+					<input readonly="readonly" type="text" class="form-control" name="${attr('name')}" placeholder="${_placeholder}" 
+						ng-model="${attr('ng-model')}" yk-read-only="${_readOnly}" ng-change="${_onChange}">
+					</input>
+					<span class="input-group-addon"><i></i></span>
+				</div>
+			</div>
+		</content>
+
+		<post-script>
+			var element = $(context.$result).find("div.colorpicker-component");
+			element = $(element);
+			
+			element.colorpicker({format: 'hex'});
+			
+			<!-- event handler for color change -->
+			var eventHandler = $.proxy(function(event){	
+			
+				var colorVal =  $(this.element.find("input")).val();
+				eval("this.$scope." + this.field + "=colorVal");
+				
+			}, {"element": element, 
+				"$scope" : context.$scope, 
+				"field": context.attr('ng-model')
+			});
+			
+			element.on('changeColor', eventHandler);
+			
+			<!-- disable the color picker -->
+			if(context.attr('yk-read-only') == 'true')
+			{
+				element.colorpicker('disable');
+			}
+			
+			<!-- watch function -->
+			var watchFun = $.proxy(function(){

```

Comment:
In watch and response functions, you are assuming the model field name starts with "model.", which may not be true. And unnecessary complexity is added.

Dont split the field. Instead directly use field name (eg: model.color) in eval. This may throw exception in cases where model is not set. To handle such cases enclose eval with try-catch and ignore such errors and proceed.